src/compile/codegen.rs

```diff
@@ -66,41 +66,45 @@ pub fn mavm_codegen<'a>(
     }
 
     let mut label_gen = LabelGenerator::new();
-    let mut code = code_in;
+    let mut code = Vec::new();
+    let mut funcs_code = BTreeMap::new();
     for func in funcs {
         if !func.imported {
-            let (lg, c) = mavm_codegen_func(
+            let id = func.name;
+            let (lg, code) = mavm_codegen_func(
```

Comment:
The overloading of the variable name `code` might confuse readers here.  It's maybe better to rename one of them.